src/atomate2/utils/powerups.py

```diff
@@ -0,0 +1,68 @@
+"""Utilities for modifying workflow."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from jobflow import Flow, Maker
+
+
+def add_metadata_to_flow(flow, additional_fields: dict, class_filter: Maker) -> Flow:
+    """
+    Return the flow with additional field(metadata) to the task doc.
+
+    This allows adding metadata to the task-docs, could be useful
+    to query results from DB.
+
+    Parameters
+    ----------
+    flow:
+    additional_fields : dict
+        A dict with metadata.
+    class_filter: .Maker
+        The Maker to which additional metadata needs to be added
+
+    Returns
+    -------
+    Flow
+        Flow with added metadata to the task-doc.
+    """
+    flow.update_maker_kwargs(
+        {"_set": {"task_document_kwargs->additional_fields": additional_fields}},
```

Comment:
Thinking about this more, this will override any additional fields already set.

I think this needs modifying slightly to 
```suggestion
        {"_set": {
            f"task_document_kwargs->additional_fields->{k}": v
            for k, v in additional_fields.items()
            }
        }
```

Reply:
Oh yes!! thanks for pointing it out. I will update this as per your suggestion 😃

Reply:
Done